src/test/requestUser.test.js

```diff
@@ -0,0 +1,263 @@
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", toUserID)
+    );
+    expect(collection).toHaveBeenCalledWith(tableName);
+  });
+
+  it("test 2: Add fromUserID to pending list of toUserID.", async () => {
+    const toUserID = "12345";
+    const fromUserID = "ABCDE";
+
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              user_id: "12345",
+              pending_list: ["Walter", "Joe"],
+              accepted_list: ["White"],
+            },
+          ],
+        });
+      },
+      empty: (fn) => true,
+    };
+    getDocs.mockImplementation(() => querySnapshot);
+
+    // Call the tested function.
+    await addUserRequest(toUserID, fromUserID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", toUserID)
+    );
+    expect(addDoc).toHaveBeenCalledWith(collection(tableName), {
+      user_id: toUserID,
+      accepted_list: [],
+      pending_list: [fromUserID],
+    });
+    expect(collection).toHaveBeenCalledWith(tableName);
+  });
+
+  it("test 1: Add user to accepted list both users.", async () => {
+    const toUserID = "12345";
+    const fromUserID = "ABCDE";
+
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              user_id: "12345",
+              pending_list: ["Walter", "Joe"],
+              accepted_list: ["White"],
+            },
+          ],
+        });
+      },
+    };
+    getDocs.mockImplementation(() => querySnapshot);
+
+    // Call the tested function.
+    await acceptRequest(toUserID, fromUserID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", toUserID)
+    );
+    expect(collection).toHaveBeenCalledWith(tableName);
+  });
+
+  it("test 2: Add user to accepted list both users.", async () => {
+    const toUserID = "12345";
+    const fromUserID = "ABCDE";
+
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              user_id: "12345",
+              pending_list: ["Walter", "Joe"],
+              accepted_list: ["White"],
+            },
+          ],
+        });
+      },
+      empty: (fn) => true,
+    };
+    getDocs.mockImplementation(() => querySnapshot);
+
+    // Call the tested function.
+    await acceptRequest(toUserID, fromUserID);
+
+    // Check that the mocked 'addDoc' and 'collection' functions were called correctly.
+    expect(query).toHaveBeenCalledWith(
+      collection(tableName),
+      where("user_id", "==", toUserID)
+    );
+    expect(addDoc).toBeCalledTimes(2);
```

Comment:
is it possible for this to be a bit more detailed to make sure it's called once for each user id, instead of just twice overall?